website/blog/2022-10-18-how-to-design-metrics.md

```diff
@@ -0,0 +1,367 @@
+
+As a longtime dbt Community member, I knew I had to get involved when I first saw the dbt Semantic Layer. It gave me a vision of a world where metrics and business logic were unified across an entire organization; a world where the data team was no longer bound to a single consuming experience and could enable their stakeholders in dozens of different ways. To me, it felt like the opportunity to contribute to the next step of what dbt could become.
+
+In past roles, I’ve been referred to as the `dbt zealot` and I’ll own that title! But it’s not a surprise - dbt was built to serve analytics engineers exactly like me. It gave us flexibility and power to serve our organizations. But it always felt like dbt best served **us.**
+
+The Semantic Layer takes this to the next level **by coupling dbt’s mature data modeling framework with semantic definitions.** The result is a first of its kind data experience that serves both the analytics engineers and stakeholders. Metrics are the first step towards this vision, allowing users to version control and centrally define their key business metrics in a single repo **while also** serving them to the entire business.
+
+However, this is still a relatively new part of the dbt tool box and you probably have a lot of questions on how **exactly** you can do that. Contained within this blog are our early recommendations for how to design and structure metrics, along with examples so that you can apply the same principles to your use cases. They were developed by combining the overall philosophy of dbt with our hands on learning gathered during the beta period and internal testing.
+
+## How to design your metrics
+
+**Pre-reading:** We recommend reading through the [metrics documentation](/docs/building-a-dbt-project/metrics), which contains a table of all the required/optional properties. 
+
+### When to put business logic in the semantic layer vs the modeling layer
+
+Our instinct when designing metrics might be to encode as much as possible into the semantic layer. An example of this is case statements - the analytics engineer’s gut instinct might be to mimic tools of the past and provide complicated case statements for the metric `expression` property to try and capture the nuance of how it should be calculated. 
+
+With the coupling of dbt’s modeling framework, this isn’t necessarily the right answer. In situations with more complicated logic, it might make more sense to keep this logic contained within the model and then reference it in the semantic layer.
+
+Getting the balance right has been a learning experience for our team as well. It’s easy to fall into comfortable patterns that might not be the best solution - it’s important for us to take a step back and think from first principles about the line of separation between what should live in a model and what should live in a metric. To explore this, we’ll walk through two examples of handling this divide.
+
+### Basic example - designing your metrics
```

Comment:
I think we can tweak the headings here to make the sectioning clearer by tweaking our H2s and H3s.

## The power of a semantic layer on top of a mature data modeling framework
### 
When to put business logic in the semantic layer vs the modeling layer

## Basic example - revenue
### Designing your metric
### logic in the modeling layer vs the semantic layer
### choosing which dimensions to include on the metric

## advanced example - NPS
### designing a complex metric
### option 1 - keeping everything in the semantic layer
### option 2 - shift logic left into the modelling layer

## How to structure your metrics
### Folder structure 
### File structure 

## A call to action